src/foam/u2/CheckBox.js

```diff
@@ -30,7 +30,7 @@ foam.CLASS({
     {
       class: 'Boolean',
       name: 'showLabel',
-      value: true
+      expression: function(label) { return !!label },
```

Comment:
Couldn't we just check that the label is defined below when we do if ( this.showLabel ), since it doesn't help that this is an expression, since we're not going to go back and re-eval the 'if' anyway?

Reply:
Good point. Alternatively, I could add a slot on that property to initE so it handles the property changing. Do you have a preference?

Reply:
Just change this to a factory. Exact same result, but less expensive. Then the user can set showLabel explicitly if they want.

Reply:
Done